packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -361,13 +361,82 @@ const NumberPrimitiveInstanceValueEditor = observer(
       resetValue,
       setValueSpecification,
     } = props;
+
```

Comment:
@MauricioUyaguari I'm a little upset how complicated this gets. I think this is not the only place and really the pain is due to us sing mobx to control the graph and the state of the component, maybe we need to think of another way to do this